Fetch video count in playlist metadata

IcySnex opened the pull request:
Closes #356
// Provided solution wasn't actually a solution. Just a workaround.

---

#### Added support for "vidoes count" property for playlist fetching:
Knowing the estimate videos count provided by the offical YouTube API without having to reiterate over the entire video collection is needed (especially when there are hundreds of videos in the playlist)
- Added `int VideosCount` to `Playlist`
- Added `int? VideosCount` to `IPlaylistData`, `PlaylistBrowseResponse`, `PlaylistNextResponse`
- Updated `PlaylistSpecs`-Tests

Tyrrrz commented:
Thank you. I'm not sure I like that some playlists will have `null` count with this approach.

IcySnex commented:
During my testing I didn't find any playlists where YouTube doesn't return the number of videos, but for safety reasons I made it nullable internally and set it to 0 in the `PlaylistClient` if it is null, similar to the `Description` property

Tyrrrz commented on YoutubeExplode.Tests/PlaylistSpecs.cs:
```diff
@@ -82,6 +83,7 @@ public async Task I_can_get_the_metadata_of_any_available_playlist(string playli
         playlist.Url.Should().NotBeNullOrWhiteSpace();
         playlist.Title.Should().NotBeNullOrWhiteSpace();
         playlist.Description.Should().NotBeNull();
+        playlist.VideosCount.Should().NotBe(null);
```
`VideosCount` is always not-`null`, so maybe `Should().NotBe(0)`?

Tyrrrz commented:
It looks like the `PlaylistNext` response parser is hard-codded to return `null`, and we use that primarily for dynamic playlists (such as mixes) and other playlists that we can't fetch normally.

What video count does it return for the non-`PL` playlists from this list? https://github.com/Tyrrrz/YoutubeExplode/blob/9f1bffdf4888b4f1e572d4e65aecee1d46fb0439/YoutubeExplode.Tests/TestData/PlaylistIds.cs#L5-L14

IcySnex commented:
I have implemented the parsing of the video count for the `PlaylistNextResponse`. I have tested the non-`PL` playlists with this and
most of them work just fine but there are minor issues with `PlaylistIds.VideoMix` & `PlaylistIds.MusicMix`.

I have inspected the YouTube API response and I have noticed that the `ContentRoot` also provides a property called `isInfinite`. If this property is true (which is the case for these Mix-Playlists) then YouTube won't tell the actual video count.

Thus I have added the `IsInfinite`-property to the `PlaylistNextResponse`. But I'm not quite sure what to do with the `VideoCount`-property in this case. Currently It will be set to `int.MaxValue` but I'm not quite sure about this approach:
https://github.com/Tyrrrz/YoutubeExplode/blob/0e6400acb6a5e07b56c9766d2f3311285a13b662/YoutubeExplode/Bridge/PlaylistNextResponse.cs#L37-L58

Tyrrrz commented:
Let's make the `VideoCount` property nullable and have `null` indicate an infinite playlist (explain so in the `<remarks>` block).

IcySnex commented:
Alright that sounds good. I have implemented it and updated the tests.

Tyrrrz commented:
Thank you :)
